src/mainwindow.cpp

```diff
@@ -2319,6 +2339,54 @@ void MainWindow::on_actionExport_Stitched_Map_Image_triggered() {
     showExportMapImageWindow(true);
 }
 
+void MainWindow::on_actionImport_Map_from_Advance_Map_1_92_triggered(){
+    importMapFromAdvanceMap1_92();
+}
+
+void MainWindow::importMapFromAdvanceMap1_92()
+{
+    QString filepath = QFileDialog::getOpenFileName(
+                this,
+                QString("Import Map from Advance Map 1.92"),
+                this->editor->project->root,
+                "Advance Map 1.92 Map Files (*.map)");
+    if (filepath.isEmpty()) {
+        return;
+    }
+
+    MapParser parser;
+    bool error = false;
+    MapLayout *mapLayout = parser.parse(filepath, &error, editor->project);
+    if (error) {
+        QMessageBox msgBox(this);
+        msgBox.setText("Failed to import map from Advance Map 1.92 .map file.");
+        QString message = QString("The .map file could not be processed. View porymap.log for specific errors.");
+        msgBox.setInformativeText(message);
+        msgBox.setDefaultButton(QMessageBox::Ok);
+        msgBox.setIcon(QMessageBox::Icon::Critical);
+        msgBox.exec();
+        return;
+    }
+
+    // TODO: This is crude because it makes a history entry for every newly-imported metatile.
```

Comment:
Can all this commented out code be removed?  Doesn't the map importer window add a brand new map to the project?  Why is this code trying to do things with metatile history?

Reply:
Sure. I believe I was using the importing metatile .bvd trigger code as a base which may have had this.